libc/benchmarks/gpu/LibcGpuBenchmark.cpp

```diff
@@ -105,8 +106,34 @@ void print_results(Benchmark *b) {
       << " stddev (num threads: " << num_threads << ")\n";
 }
 
+// We want our random values to be approximately
+// |real value| <= 2^(max_exponent) * (1 + (random 52 bits) * 2^-52) <
+// 2^(max_exponent + 1)
+// The largest integer that can be stored in a double is 2^53
+static constexpr int MAX_EXPONENT = 52;
+
+static double get_rand() {
+  using FPBits = LIBC_NAMESPACE::fputil::FPBits<double>;
+  uint64_t bits = LIBC_NAMESPACE::rand();
+  double scale = 0.5 + MAX_EXPONENT / 2048.0;
+  FPBits fp(bits);
+  fp.set_biased_exponent(
+      static_cast<uint32_t>(fp.get_biased_exponent() * scale));
+  return fp.get_val();
+}
+
+static void init_random_input() {
+  LIBC_NAMESPACE::srand(LIBC_NAMESPACE::gpu::processor_clock());
+  for (int i = 0; i < RANDOM_INPUT_SIZE; i++) {
+    random_input[i] = get_rand();
+  }
+}
+
```

Comment:
I figured this would be handled in the benchmark itself, like we do it before the latency calculation.

Reply:
Ah, yeah that makes sense. I was thinking it might reduce duplication if we have to do the same thing for multiple benchmarks, but I think you're right about moving it to the benchmark itself

Reply:
But now that I think about it, what was the way to share the array between all threads without using a global? I'm assuming that we want all threads to see the same random input, right?

Reply:
Can put it as a utility in the header.

Reply:
Do you mean keep the array in the header but call `init_random_array()` from the benchmark itself?

Reply:
Yeah I just mean if you want a utility to get some random input, just make it a function in the header and have the benchmarks call it.